test/unit/io/alignment_file/format_sam_test.cpp

```diff
@@ -117,10 +119,457 @@ struct alignment_data : public ::testing::Test
         sam_tag_dictionary{},
         sam_tag_dictionary{}
     };
+};
+
+// ----------------------------------------------------------------------------
+// reading
+// ----------------------------------------------------------------------------
+
+TEST(read_header, sucess)
+{
+    std::string header_str
+    {
+        "@HD\tVN:1.0\tSO:coordinate\tSS:coordinate:queryname\tGO:none\n"
+        "@PG\tID:qc\tPN:quality_control\tCL:qc -f file1\tDS:trim reads with low qual\tVN:1.0.0\n"
+        "@PG\tID:novoalign\tPN:novoalign\tVN:V3.02.07\tCL:novoalign -d /path/hs37d5.ndx -f /path/file.fastq.gz -F STDFQ"
+        " --Q2Off -t 400 -o SAM -c 10\tPP:qc\n"
+        "@SQ\tSN:1\tLN:249250621\n"
+        "@SQ\tSN:2\tLN:243199373\tAS:hs37d5\n"
+        "@RG\tID:U0a_A2_L1\tPL:illumina\tPU:1\tLB:1\tSM:NA12878\n"
+        "@RG\tID:U0a_A2_L2\tPL:illumina\tSM:NA12878\tPU:1\tLB:1\n"
+        "@CO\tTralalalalalala this is a comment\n"
+        "read1\t41\tref\t1\t61\t1S1M1D1M1I\tref\t10\t300\tACGT\t!##$\tAS:i:2\tNM:i:7\n"
```

Comment:
raw string?

Reply:
My editor always converts tabs to spaces so a raw string is tedious :(

Reply:
"My dog ate the homework, that's why it's impossible to read" :wink: 

I am pretty sure you can turn that behaviour off!

Reply:
:roll_eyes: no because in general I want this behaviour (I will not make four spaces everytime I indent when coding) and I don't want to switch every time.. and many people have this, so if they open and save this file there will always be changes...

Reply:
> and I don't want to switch every time

I think switching once for some test file every month takes less time than us arguing right now :smile_cat: 

> and many people have this, so if they open and save this file there will always be changes...

Opening and saving shouldn't affect those lines, at least with every sensible editor I have seen. We have raw string with tabs in other places, too. It really makes reading the stuff a lot easier .

Reply:
> I think switching once for some test file every month takes less time than us arguing right now 

we agree on that point :smile_cat: 

I'm still against it but I will change it..